src/core/batch/batch.ts

```diff
@@ -27,25 +27,23 @@ interface BatchFileData {
   hash: HexString;
 }
 
-type SignedAnnouncementIterable<T extends AnnouncementType> = AsyncOrSyncIterable<
-  AnnouncementWithSignature<TypedAnnouncement<T>>
->;
+type SignedAnnouncementIterable<T extends AnnouncementType> = AsyncOrSyncIterable<AnnouncementWithSignature<T>>;
 
-type ParquetRecord = Record<string, Buffer | Uint8Array | BigInt | number>;
+type ParquetRecord = Record<string, Buffer | Uint8Array | bigint | number>;
 
 const parseAnnouncement = <T extends SignedAnnouncement>(record: ParquetRecord): T => {
-  const schema = getSchemaFor(Number(record.announcementType));
-  const announcement: Record<string, string | number | BigInt> = {};
+  const schema = getSchemaFor(record.announcementType as AnnouncementType);
 
-  for (const key in schema) {
-    if (schema[key].type === "BYTE_ARRAY") {
-      announcement[key] = record[key].toString();
+  const announcement = Object.entries(schema).reduce<Record<string, string | number | bigint>>((acc, [key, s]) => {
```

Comment:
The types really didn't like working with schema and announcement in this way, so I switched over to a reduce setup.